extensions/websockets-next/server/runtime/src/main/java/io/quarkus/websockets/next/WebSocket.java

```diff
@@ -0,0 +1,55 @@
+package io.quarkus.websockets.next;
+
+import static java.lang.annotation.ElementType.TYPE;
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
+
+import io.smallrye.common.annotation.Experimental;
+
+/**
+ * Denotes a WebSocket endpoint.
+ */
+@Retention(RUNTIME)
+@Target(TYPE)
+@Experimental("This API is experimental and may change in the future")
+public @interface WebSocket {
+
+    /**
+     * The path of the endpoint.
+     * <p>
+     * It is possible to match path parameters. The placeholder of a path parameter consists of the parameter name surrounded by
+     * curly brackets. The actual value of a path parameter can be obtained using
+     * {@link WebSocketServerConnection#pathParam(String)}. For example, the path <code>/foo/{bar}</code> defines the path
+     * parameter {@code bar}.
+     *
+     * @see WebSocketServerConnection#pathParam(String)
+     */
+    public String path();
```

Comment:
I would prefer that this were `value` instead of `path`, but I can live with it :)

Reply:
My problem with `value()` is that once you add more annotation methods it becomes unclear, e.g. `@WebSocket(value = "/foo", executionMode = CONCURRENT)`. But we can discuss this later...

Reply:
Yeah I know, but the most common case is to just add the path.

> But we can discuss this later

Of course